src/components/CarouselDots.js

```diff
@@ -27,20 +27,26 @@ export default class CarouselDots extends Component {
 
   renderCarouselDots() {
     if (this.props.thumbnails) {
-      return this.props.thumbnails.map((thumbnail, index) => (
```

Comment:
this could be done simpler:

```
const dotsLength = isNaN(this.props.number) ? this.props.thumbnails.length : this.props.number;
return this.props.thumbnails.slice(dotsLength).map((thumbnail, index) => (
```